### PR TITLE
Add new tests for seven modules. Include GH workflow.

### PR DESCRIPTION
**Pull request by lbonaldo**
## Description

This PR adds seven more tests that are executed whenever a pull request is `opened`, `synchronize`, and `reopened`, or at every push targeting `main` or `develop.` The tests are:

- `test_threezones.jl`: tests GenX over **three zones** and five representative days;
- `test_multistage.jl`: tests the **multistage** module over a single zone and five representative days;
- `test_electrolyzer.jl`: tests the **electrolyzer** module over five representative days; 
- `test_VREStor.jl`: tests **VRE** and **storage** over 24 hours;
- `test_piecewisefuel_CO2.jl`: tests the **piecewise fuel CO2** module over 24 hours; 
- `test_time_domain_reduction.jl`: tests the **TDR** module;
- `test_methodofmorris.jl`: tests the **Method of Morris** module.

For the first five cases, the tests are passed if the value of the objective function is consistent over time (up to the tolerance of the solver). The results are stored in a log folder updated in a separate repository (`GenXProject/GenX-testlog.git`). 
The tests in `test_time_domain_reduction.jl` validate the clustering for the TDR module by computing the [*rand index*](https://en.wikipedia.org/wiki/Rand_index) and the [*mutual information*](https://en.wikipedia.org/wiki/Mutual_information) and by comparing the output files. 
For the `Method of Morris` module, we are testing if the module is correctly built, and if the output is consistent over time.

I'd appreciate any critiques or suggestions on how to improve these tests.

## What type of PR is this?

- [x] Feature

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and .md files under /docs/src have been updated if necessary.
- [x] The latest changes on the target branch have been incorporated, so that any conflicts are taken care of before merging. This can be accomplished either by merging in the target branch (e.g. 'git merge develop') or by rebasing on top of the target branch (e.g. 'git rebase develop'). Please do not hesitate to reach out to the GenX development team if you need help with this.
- [x] Code has been tested to ensure all functionality works as intended.
- [x] CHANGELOG.md has been updated (if this is a 'notable' change).
- [x] I consent to the release of this PR's code under the GNU General Public license.

## How this can be tested

After checking out `develop`, enter the package manager and run `test` or `test GenX`. The tests should run automatically.

## Post-approval checklist for GenX core developers

After the PR is approved

- [x] Check that the latest changes on the target branch are incorporated, either via merge or rebase
- [x] Remember to squash and merge if incorporating into develop

### CODE REVIEW

**Review by cfe316 — APPROVED** [review]
This looks great. My comments are mostly cosmetic, except for the possible interactions with Gabe's commerical-solver-use PR (see test/utilities.jl). Approving because I think this should still work.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  test:
+    name: Julia ${{ matrix.version }} - ${{ matrix.os }} - ${{ matrix.arch }} - ${{ github.event_name}}
+    runs-on: ${{ matrix.os }}
+    strategy: 
+      fail-fast: false
+      matrix:
+        version:
+          - "1.6"
+          - "1.7"
+          - "1.8"
```
[the remark this answers is not in the document]
Comment (lbonaldo) [reply .github/workflows/ci.yml:19]: Yes, `'1'` ensures we're testing on the latest stable release. We could also keep only the current stable release (`'1'`), and the long-term support (`LTS`) release (right now, v1.6.7) as other packages do.

File: test/test_threezones.jl [hunk 2]
```diff
@@ -0,0 +1,52 @@
+module TestThreeZones
+
+using Test
+
+include(joinpath(@__DIR__, "utilities.jl"))
+
+obj_true = 6.9602085499e+03
```
[the remark this answers is not in the document]
Comment (lbonaldo) [reply test/test_threezones.jl:7]: I run the example several times and check if the value of the objective function was stable. 
You're right, thanks, I will round them to match the tolerance (I'm using `Optimal_Tol: 1.0e-05 `)

File: test/Electrolyzer/Period_map.csv [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+23,72,2
+24,72,2
+25,72,2
+26,72,2
+27,27,1
+28,72,2
+29,72,2
+30,72,2
+31,72,2
+32,72,2
```
Comment (cfe316) [inline test/Electrolyzer/Period_map.csv:33]: This Period_map file seems odd. Is this 365 days being represented by 5 days?

Comment (lbonaldo) [reply test/Electrolyzer/Period_map.csv:33]: Yes.

**Pull request merged**